src/graph/state.cc

```diff
@@ -142,6 +142,25 @@ namespace hpp {
         hppDout (error, "Edge not found");
         return 0;
       }
+
+      void State::addNumericalConstraint(const ImplicitPtr_t& numConstraint)
+      {
+        ComparisonTypes_t comp(numConstraint->comparisonType());
+        for(constraints::ComparisonType c : comp)
+        {
+          if (c == constraints::Equality)
+          {
+            std::ostringstream os;
+            os << "Failed to insert constraint \""
+               << numConstraint->function().name()
+               << "\" as a state constraint since it contains a comparison "
+              "of type Equality";
+            throw std::logic_error(os.str().c_str());
```

Comment:
FYI, when you only have string, you don't have to use ostringstream. The operator+ is enough.